### ISSUE TITLE
Improve API of Attribute fields 

### ISSUE THREAD

**Issue by math-fehr** [xdslproject/xdsl#535]
Currently, an `IntegerAttr` looks like (roughly)
```
class IntegerAttr:
    value: IntAttr
    type: IntegerType
```
    
We could change this to 
```
class IntegerAttr:
    value_attr: IntAttr
    type: IndexType | IntegerType

    @property
    def value():
        return value_attr.data

    @property
    def type_width():
        return type.width.data
```

Essentially, changing the "default" field name to directly get the Python value, rather than the underlying Attribute.

**Comment by webmiche** [comment]
I am strongly guessing that you propose changing the decorator to simulate this behavior.

**Comment by math-fehr** [comment]
Oh no, it's just changing attribute definitions, and adding appropriate properties.

**Comment by webmiche** [comment]
By the user by hand?

**Comment by math-fehr** [comment]
Yes, we have currently no way to define new fields without PyRight complaining.
This is costly, yes, but this is so much better than writing `my_integer.value.data` every single time.

**Comment by math-fehr** [comment]
If we could autogenerate it, that would be great though ;)

**Comment by webmiche** [comment]
How about two stage definition, TableGen-style: We generate python files from the dialect definitions and let pyright look at those instead of the python-embedded DSL.

**Comment by webmiche** [comment]
This is a broader topic than this though, I guess :)

**Comment by compor** [comment]
I wouldn't mind a more xDSL-y name, but I feel this disruption is unnecessary now (esp. for downstream clients).
Nothing that an `import ... as` cannot resolve really, but yeah, I'm in favour of closing.

**Comment by superlopuh** [comment]
Closing
